sfs/time/soundfield.py

```diff
@@ -1,48 +1,57 @@
 """Compute sound field."""
 
+from __future__ import division
 import numpy as np
-
 from .. import util
-from . import source
 from .. import defs
-
-
-def synthesize_p(d_line, channel_weight, x0, grid, t, fs=None):
-    """Compute sound field for given time sample.
-
-    D_line(m x n) specifies the signals m fed into each secondary source n.
-    Channel_weight is applied during integration, e.g. source tapering.
-    x0 are available secandary sources.
+from .source import point
+
+
+def p_array(x0, d, channel_weights, t, grid, source=point, fs=None, c=None):
+    """Compute sound field for an array of secondary sources for a given time.
+
+    Parameters
+    ----------
+    x0 : (N, 3) array_like
+        Sequence of secondary source positions.
+    d : (N, C) numpy.ndarray
+        Specifies the signals N(t) fed into each secondary source C (columns).
+        "N" is signal samples(t) per channel, and
+        "C" is channels
+    channel_weights : (C,) array_like
+        Additional weights applied during integration, e.g. source tapering.
+    t : float
+        Simulation point in time (seconds).
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    source: function, optional
+        Source type is a function, returning scalar field.
+        For default, see `sfs.time.source.point()`
+    fs: int, optional
+        Sampling frequency in hertz.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Pressure at grid positions.
 
     """
     if fs is None:
         fs = defs.fs
-    channel_weight = util.asarray_1d(channel_weight)
-    if len(channel_weight) != len(x0):
+    if c is None:
+        c = defs.c
+    x0 = util.asarray_of_rows(x0)
```

Comment:
Sure, no problem.

Just for the record, it could be very similar to `asarray_of_rows()`, there's really only one character moved to a different place:

```python
def asarray_of_columns(a, **kwargs):
    result = np.squeeze(np.asarray(a, **kwargs))
    if result.ndim == 1:
        result = result.reshape(-1, 1)
    return result
```